uncertainty_baselines/models/segmenter_be.py

```diff
@@ -0,0 +1,211 @@
+  patches: ml_collections.ConfigDict
+  backbone_configs: ml_collections.ConfigDict
+  decoder_configs: ml_collections.ConfigDict
+  head_kernel_init: InitializeFn = nn.initializers.zeros
+
+  @nn.compact
+  def __call__(self, x: jnp.ndarray, *, train: bool, debug: bool = False):
+    input_shape = x.shape
+    b, h, w, _ = input_shape
+
+    fh, fw = self.patches.size
+    gh, gw = h // fh, w // fw
+
+    if self.backbone_configs.type == 'vit' and self.decoder_configs.type == 'linear':
+      assert self.backbone_configs.ens_size == 1
+
+    if self.backbone_configs.type == 'vit' and self.decoder_configs.type == 'linear_be':
+      raise NotImplementedError('Configuration with encoder {} and decoder {} is not implemented'.format(
+          self.backbone_configs.type,
+          self.decoder_configs.type,
+      ))
+
+    if self.backbone_configs.type == 'vit':
+      x, out = segmenter.ViTBackbone(
+          mlp_dim=self.backbone_configs.mlp_dim,
+          num_layers=self.backbone_configs.num_layers,
+          num_heads=self.backbone_configs.num_heads,
+          patches=self.patches,
+          hidden_size=self.backbone_configs.hidden_size,
+          dropout_rate=self.backbone_configs.dropout_rate,
+          attention_dropout_rate=self.backbone_configs.attention_dropout_rate,
+          classifier=self.backbone_configs.classifier,
+          name='backbone')(
+              x, train=train)
+    elif self.backbone_configs.type == 'vit_be':
+      # import pdb; pdb.set_trace()
+      x, out = ViTBackboneBE(
+          mlp_dim=self.backbone_configs.mlp_dim,
+          num_layers=self.backbone_configs.num_layers,
+          num_heads=self.backbone_configs.num_heads,
+          patches=self.patches,
+          hidden_size=self.backbone_configs.hidden_size,
+          dropout_rate=self.backbone_configs.dropout_rate,
+          attention_dropout_rate=self.backbone_configs.attention_dropout_rate,
+          classifier=self.backbone_configs.classifier,
+          ens_size=self.backbone_configs.ens_size,
+          random_sign_init=self.backbone_configs.random_sign_init,
+          be_layers=self.backbone_configs.be_layers,
+          name='backbone')(
+              x, train=train)
+    else:
+      raise ValueError(f'Unknown backbone: {self.backbone_configs.type}.')
+
+    if self.decoder_configs.type == 'linear':
+      output_projection = nn.Dense(
+        self.num_classes,
+        kernel_init=nn.initializers.zeros,
+        name='output_projection')
+    elif self.decoder_configs.type == 'linear_be':
+      output_projection = ed.nn.DenseBatchEnsemble(
+          self.num_classes,
+          self.backbone_configs.ens_size,
+          activation=None,
+          alpha_init=ed.nn.utils.make_sign_initializer(
+              self.backbone_configs.get("random_sign_init")),
+          gamma_init=ed.nn.utils.make_sign_initializer(
+              self.backbone_configs.get("random_sign_init")),
+          kernel_init=self.head_kernel_init,
```

Comment:
`kernel_init` should be consistent with linear case, so you can do just the zeros here or set it as a config in both.